### PR TITLE
passing components folder to daprd

### PR DESCRIPTION
**Pull request by amanbha**
Fixes https://github.com/dapr/cli/issues/366

### CODE REVIEW

**Comment by yaron2** [comment]
https://github.com/dapr/cli/blob/737a4e6e776466e772db9b3a1c91dd4bbdbad1bd/pkg/standalone/run.go#L47

**Comment by yaron2** [comment]
Need to make sure that field gets populated with the default dir path if not supplied by the user.

File: pkg/standalone/run.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ type componentMetadataItem struct {
 	Value string `yaml:"value"`
 }
 
-func getDaprCommand(appID string, daprHTTPPort int, daprGRPCPort int, appPort int, configFile, protocol string, enableProfiling bool, profilePort int, logLevel string, maxConcurrency int, placementHost string) (*exec.Cmd, int, int, int, error) {
+func getDaprCommand(appID string, daprHTTPPort int, daprGRPCPort int, appPort int, configFile, protocol string, enableProfiling bool, profilePort int, logLevel string, maxConcurrency int, placementHost string, componentsPath string) (*exec.Cmd, int, int, int, error) {
```
Comment (youngbupark) [inline pkg/standalone/run.go:76]: is componentsPath always current directory ?

**Comment by amanbha** [comment]
> Need to make sure that field gets populated with the default dir path if not supplied by the user.

Thats what the code is doing, its getting the default path 

> There's a components path field on the Config struct

Thats only set when user overrides it. Here its using the default path when not overridden by user, see here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273

File: pkg/standalone/run.go (hunk 1 above)
Comment (amanbha) [reply pkg/standalone/run.go:76]: its set here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273

Comment (yaron2) [reply pkg/standalone/run.go:76]: I don't see where it is set, @amanbha. Trace the call from from cmd/run

**Comment by youngbupark** [comment]
> > Need to make sure that field gets populated with the default dir path if not supplied by the user.
> 
> Thats what the code is doing, its getting the default path
> 
> > There's a components path field on the Config struct
> 
> Thats only set when user overrides it. Here its using the default path when not overridden by user, see here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273

link may be wrong. https://github.com/dapr/cli/blob/c37118166b1b6c869cd2795863eb99f4960fe100/cmd/run.go#L248

this is not the current directly. 

the code always set componentsPath.. But I cannot find where componentsPath is set by default.

File: pkg/standalone/run.go (hunk 1 above)
Comment (yaron2) [reply pkg/standalone/run.go:76]: We need to make sure `componentsPath` is set to the default path when it's passed to this method.

Comment (yaron2) [reply pkg/standalone/run.go:76]: This is missing a default:

https://github.com/dapr/cli/blob/c37118166b1b6c869cd2795863eb99f4960fe100/cmd/run.go#L248

Comment (yaron2) [reply pkg/standalone/run.go:76]: We're telling the user there's a default  path, but its not being set.

**Comment by yaron2** [comment]
> > > Need to make sure that field gets populated with the default dir path if not supplied by the user.
> > 
> > 
> > Thats what the code is doing, its getting the default path
> > > There's a components path field on the Config struct
> > 
> > 
> > Thats only set when user overrides it. Here its using the default path when not overridden by user, see here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273
> 
> link may be wrong.
> 
> https://github.com/dapr/cli/blob/c37118166b1b6c869cd2795863eb99f4960fe100/cmd/run.go#L248
> 
> this is not the current directory.
> 
> the code always set componentsPath.. But I cannot find where componentsPath is set by default.

Indeed. See my comments above.

File: pkg/standalone/run.go (hunk 1 above)
Comment (yaron2) [reply pkg/standalone/run.go:76]: The default dir `getDefaultComponentsFolder()` should be set as the default value for the flag here.

**Comment by amanbha** [comment]
> > > > Need to make sure that field gets populated with the default dir path if not supplied by the user.
> > > 
> > > 
> > > Thats what the code is doing, its getting the default path
> > > > There's a components path field on the Config struct
> > > 
> > > 
> > > Thats only set when user overrides it. Here its using the default path when not overridden by user, see here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273
> > 
> > 
> > link may be wrong.
> > https://github.com/dapr/cli/blob/c37118166b1b6c869cd2795863eb99f4960fe100/cmd/run.go#L248
> > 
> > this is not the current directory.
> > the code always set componentsPath.. But I cannot find where componentsPath is set by default.
> 
> Indeed. See my comments above.

Adding the correct link, componentsPath is getting set here https://github.com/dapr/cli/blob/fix_366/pkg/standalone/run.go#L273

which is calling func at https://github.com/dapr/cli/blob/fix_366/pkg/standalone/run.go#L347

**Comment by yaron2** [comment]
> > > > > Need to make sure that field gets populated with the default dir path if not supplied by the user.
> > > > 
> > > > 
> > > > Thats what the code is doing, its getting the default path
> > > > > There's a components path field on the Config struct
> > > > 
> > > > 
> > > > Thats only set when user overrides it. Here its using the default path when not overridden by user, see here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273
> > > 
> > > 
> > > link may be wrong.
> > > https://github.com/dapr/cli/blob/c37118166b1b6c869cd2795863eb99f4960fe100/cmd/run.go#L248
> > > 
> > > this is not the current directory.
> > > the code always set componentsPath.. But I cannot find where componentsPath is set by default.
> > 
> > 
> > Indeed. See my comments above.
> 
> Adding the correct link, componentsPath is getting set here https://github.com/dapr/cli/blob/fix_366/pkg/standalone/run.go#L273
> 
> which is calling func at https://github.com/dapr/cli/blob/fix_366/pkg/standalone/run.go#L347

Can we use the CLI flags to set default values? I want to keep it consistent.

**Comment by amanbha** [comment]
> > > > > > Need to make sure that field gets populated with the default dir path if not supplied by the user.
> > > > > 
> > > > > 
> > > > > Thats what the code is doing, its getting the default path
> > > > > > There's a components path field on the Config struct
> > > > > 
> > > > > 
> > > > > Thats only set when user overrides it. Here its using the default path when not overridden by user, see here https://github.com/dapr/cli/pull/367/files#diff-184e85420a4841e42d77c23247296911R273
> > > > 
> > > > 
> > > > link may be wrong.
> > > > https://github.com/dapr/cli/blob/c37118166b1b6c869cd2795863eb99f4960fe100/cmd/run.go#L248
> > > > 
> > > > this is not the current directory.
> > > > the code always set componentsPath.. But I cannot find where componentsPath is set by default.
> > > 
> > > 
> > > Indeed. See my comments above.
> > 
> > 
> > Adding the correct link, componentsPath is getting set here https://github.com/dapr/cli/blob/fix_366/pkg/standalone/run.go#L273
> > which is calling func at https://github.com/dapr/cli/blob/fix_366/pkg/standalone/run.go#L347
> 
> Can we use the CLI flags to set default values? I want to keep it consistent.

Sure, we can

**Comment by yaron2** [comment]
I think you forgot to paste the function to cmd/run.go

**Pull request merged**